### DISCUSSION TITLE
MLFlow tracking - Compare correlation matrix between different runs

### DISCUSSION THREAD

**Discussion by schergui-thefork** [mlflow/mlflow#7919] [Q&A] [answered]
Hello everyone,

I have a numpy array that contains correlation number between different columns. I'd like to log it in MLFlow Tracking and compare the array between runs. Is there a way to accomplish this ?

Thanks

**Comment by JHibbard** [comment, marked as answer]
Hi schergui-thefork,

I can think of a few ways to do this but you'll have to write some helper functions to perform the comparisons, there's no built-in support for comparing logged arrays of correlation numbers:

1) Log pickled versions of the numpy arrays as artifacts. This might be best if your arrays are large, I'd suggest this.
2) Don't log the numpy array, but instead use `log_dict` and have the keys be the name/index of the columns.

You can then use the search APIs to collect runs that have a pickled numpy array artifact (give the artifacts a consistent name) or dictionary, retrieve the correlation data, and then perform your comparisons.

Hope that helps!
Jim
